test/mpi.py

```diff
@@ -23,36 +23,251 @@
+    #         raise Exception("something went wrong while trying to load OpenMPI module: %s" % loadoutput)
+
+    #     instance = getinstance(OpenMPI, Local, optionparser)
+
+    #     print("mpiscriptname: %s, path: %s, instance mpirun for: %s" %
+    #           (instance._mpiscriptname_for, mpim.which('mpirun'),
+    #            instance._mpirun_for))
+    #     self.assertTrue(instance._is_mpirun_for(mpim.which('mpirun')),
+    #                     msg="mpi instance is not an MPI flavor defined by %s according to _is_mpirun_for, path: %s" %
+    #                     (OpenMPI, mpim.which('mpirun')))
+
+    #     exitcode1, _ = run_simple("module purge")
+    #     exitcode2, _ = run_simple("module load cluster/delcatty") # load default cluster\
+    #     if exitcode1 + exitcode2 > 0:
+    #         raise Exception("something went wrong while trying to reset loaded modules")
+
+    def test_set_omp_threads(self):
+        """test if OMP_NUM_THREAD gets set correctly"""
+        mpi_instance = getinstance(mpim.MPI, Local, MympirunOption())
+        mpi_instance.set_omp_threads()
+        self.assertTrue(getattr(mpi_instance.options, 'ompthreads') is not None, msg="ompthreads was not set")
+        self.assertEqual(os.environ["OMP_NUM_THREADS"], getattr(mpi_instance.options, 'ompthreads', None),
+                         msg="ompthreads has not been set in the environment variable OMP_NUM_THREADS")
+
+    def test_set_netmask(self):
+        """test if netmask matches the layout of an ip adress"""
+        mpi_instance = getinstance(mpim.MPI, Local, MympirunOption())
+        mpi_instance.set_netmask()
+        # matches "IP address / netmask"
+        reg = re.compile(r"\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}")
+        print("netmask: %s" % mpi_instance.netmask)
+        for substr in string.split(mpi_instance.netmask, sep=":"):
+            self.assertTrue(reg.match(substr), msg="%s does not seem to be a valid address" % substr)
+
+    def test_select_device(self):
+        """test if device and netmasktype are set and are picked from a list of options"""
+        mpi_instance = getinstance(mpim.MPI, Local, MympirunOption())
+        mpi_instance.select_device()
+        self.assertTrue(mpi_instance.device and mpi_instance.device in mpi_instance.DEVICE_MPIDEVICE_MAP.values(),
+                        msg="%s is not a valid device type, possible values: %s" %
+                        (mpi_instance.device, mpi_instance.DEVICE_MPIDEVICE_MAP.values()))
+        self.assertTrue(mpi_instance.netmasktype and mpi_instance.netmasktype in mpi_instance.NETMASK_TYPE_MAP.values(),
+                        msg="%s is not a valid netmask type, possible values: %s" %
+                        (mpi_instance.netmasktype, mpi_instance.NETMASK_TYPE_MAP.values()))
+
+    def test_make_node_file(self):
+        """test if the nodefile is made and if it contains the same amount of nodas as mpinodes"""
```

Comment:
as we do local, we can check if get the correct number of cores (==lines in nodefile)?